consensus/src/processes/pruning_proof/mod.rs

```diff
@@ -670,6 +683,26 @@ impl PruningProofManager {
         }
     }
 
+    /// Returns the k + 1 chain blocks below this hash (inclusive). If data is missing
+    /// the search is halted and a partial chain is returned.
+    ///
+    /// The returned hashes are guaranteed to have GHOSTDAG data
+    pub(crate) fn get_ghostdag_chain_k_depth(&self, hash: Hash) -> Vec<Hash> {
```

Comment:
Do we need this specifically to be just about `get_ghostdag_chain_k_depth`? A generic `get_ghostdag_chain_with_depth(&self, hash: Hash, depth: u64)` may make it useful for other possible use-cases, like if we ever need to dig deeper.